OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -631,6 +670,94 @@ void AssignRolesToIdleUnits(Actor self)
 			}
 
 			FindAndDeployBackupMcv(self);
+
+			if (--minCaptureDelayTicks <= 0)
+			{
+				minCaptureDelayTicks = Info.MinimumCaptureDelay;
+				QueueCaptureOrders();
+			}
+		}
+
+		IEnumerable<Actor> GetVisibleActorsBelongingToPlayer(Player owner)
+		{
+			foreach (var actor in GetActorsThatCanBeOrderedByPlayer(owner))
+				if (actor.CanBeViewedByPlayer(owner))
+					yield return actor;
+		}
+
+		IEnumerable<Actor> GetActorsThatCanBeOrderedByPlayer(Player owner)
+		{
+			foreach (var actor in World.Actors)
+				if (actor.Owner == owner && !actor.IsDead && actor.IsInWorld)
+					yield return actor;
+		}
+
+		void QueueCaptureOrders()
+		{
+			if (!Info.CapturingActorTypes.Any() || Player.WinState != WinState.Undefined)
+				return;
+
+			var capturers = unitsHangingAroundTheBase.Where(a => a.IsIdle && Info.CapturingActorTypes.Contains(a.Info.Name)).ToArray();
+			if (capturers.Length == 0)
+				return;
+
+			var randPlayer = World.Players.Random(Random);
```

Comment:
I didn't build out an agro system for capturing (because that isn't logical IMO) so this will cause enemies and neutral  (and allies, but the capturing code may filter that out) to be targeted without bias.